src/peer_connection.cpp

```diff
@@ -605,7 +605,8 @@ namespace libtorrent
 			for (int i = 0; i < int(hash.size() / sizeof(std::uint32_t)); ++i)
 			{
 				++loops;
-				piece_index_t const piece(detail::read_uint32(p) % num_pieces);
+				TORRENT_ASSERT(num_pieces > 0);
+				piece_index_t const piece(int(detail::read_uint32(p) % std::uint32_t(num_pieces)));
```

Comment:
do you really need the ``int`` cast though? Since you're explicitly constructing the ``piece_index_t``, I would expect you to get an implicit cast there, without issues